File: sdk/nodejs/src/resources/dns.js
"use strict";

const { BaseResource } = require("./base");

class DnsResource extends BaseResource {
  async listZones(params = {}) {
    return this._http.get("/api/dns");
  }

  async createZone(serviceId, data) {
    return this._http.post("/api/service/{service_id}/dns", data);
  }

  async getZones(serviceId) {
    return this._http.get("/api/service/{service_id}/dns");
  }

  async getZone(serviceId, zoneId) {
    return this._http.get("/api/service/{service_id}/dns/{zone_id}");
  }

  async deleteZone(serviceId, zoneId) {
    return this._http.delete("/api/service/{service_id}/dns/{zone_id}");
  }

  async createRecord(serviceId, zoneId, data) {
    return this._http.post("/api/service/{service_id}/dns/{zone_id}/records", data);
  }

  async updateRecord(serviceId, zoneId, recordId, data) {
    return this._http.put("/api/service/{service_id}/dns/{zone_id}/records/{record_id}", data);
  }

  async deleteRecord(serviceId, zoneId, recordId) {
    return this._http.delete("/api/service/{service_id}/dns/{zone_id}/records/{record_id}");
  }

}

module.exports = { DnsResource };

File: sdk/csharp/Resources/Dns.cs
namespace HiTechCloud.SDK.Resources;

public class Dns : BaseResource
{
    public Dns(HttpClientWrapper http) : base(http) { }

    public async Task<Dictionary<string, object?>> DnsServicesAsync()
    {
        return await Http.GetAsync("/api/dns");
    }

    public async Task<Dictionary<string, object?>> CreateZoneAsync(int serviceId, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{serviceId}/dns", serviceId), data);
    }

    public async Task<Dictionary<string, object?>> ZonesAsync(int serviceId)
    {
        return await Http.GetAsync(string.Format("/api/service/{serviceId}/dns", serviceId));
    }

    public async Task<Dictionary<string, object?>> ZoneAsync(int serviceId, int zoneId)
    {
        return await Http.GetAsync(string.Format("/api/service/{serviceId}/dns/{zoneId}", serviceId, zoneId));
    }

    public async Task<Dictionary<string, object?>> DeleteZoneAsync(int serviceId, int zoneId)
    {
        return await Http.DeleteAsync(string.Format("/api/service/{serviceId}/dns/{zoneId}", serviceId, zoneId));
    }

    public async Task<Dictionary<string, object?>> CreateRecordAsync(int serviceId, int zoneId, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{serviceId}/dns/{zoneId}/records", serviceId, zoneId), data);
    }

    public async Task<Dictionary<string, object?>> UpdateRecordAsync(int serviceId, int zoneId, int recordId, Dictionary<string, object> data)
    {
        return await Http.PutAsync(string.Format("/api/service/{serviceId}/dns/{zoneId}/records/{recordId}", serviceId, zoneId, recordId), data);
    }

    public async Task<Dictionary<string, object?>> DeleteRecordAsync(int serviceId, int zoneId, int recordId)
    {
        return await Http.DeleteAsync(string.Format("/api/service/{serviceId}/dns/{zoneId}/records/{recordId}", serviceId, zoneId, recordId));
    }
}
